### ISSUE TITLE
[VFS] Saving does a write twice when file exists and overwrites even when cancel the overwrite dialog.

### ISSUE THREAD

**Issue by RickGlimmer** [os-js/OS.js#301]
When overwriting a file (Textpad, writer and probably others) it will send four! requests to the server.
1. scandir (this will get the current folders contents)
2. **write** ( **this will actually overwrite the file**)
3. exists (this gives true and prompts the overwrite dialog)
4. write (when confirming **OR cancelling the overwrite**)

When saving a file with a new name
1. scandir
2. exists
3. write
4. **write**

On the client side it should be:
1 scandir
2 exists
3 write (when confirmed to overwrite or if file does not exist)

On the server side a write without the 'overwrite' option set should not overwrite a file, only after a overwrite dialog has been confirmed and a 'overwrite' option is send with the request.

There is another problem when the dialog appears and that is if you click on the directory list the dialog is hidden behind it. It has the 'on top' checked but it does not stay on top of the save dialog.

**Comment by andersevenrud** [comment]
Nice catch! Investegating

**Comment by andersevenrud** [comment]
A missing `return false` statement Oo.

**Comment by RickGlimmer** [comment]
Yes that fixed it.
I would like to see a 'overwrite' : 'true' added to prevent the backend overwriting a file without clear permission. I'll make a new issue for a improvement.

**Comment by andersevenrud** [comment]
Ok. I'll just duplicate what happens in upload function (it has `overwrite`)
